### PR TITLE
tm analytics - draft

### PR DESCRIPTION
**Pull request by alonp99**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: services/workflows-service/package.json [hunk 1]
```diff
@@ -138,6 +139,7 @@
     "jest": "29.7.0",
     "jest-mock-extended": "^2.0.4",
     "json-schema-to-typescript": "^13.0.1",
+    "libpg-query": "^15.0.2",
```
Comment (liorzam) [inline services/workflows-service/package.json:142]: are you sure you meant to install it as **dev** dependency?

File: services/workflows-service/src/data-analytics/data-analytics.repository/analytics.query-builder.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  addDateRange(timeUnit: 'minutes' | 'hours' | 'days' | 'months', amount: number) {
+    this.conditions.push(
+      Prisma.sql`transactionDate >= CURRENT_DATE - INTERVAL '${amount} ${timeUnit}'`,
+    );
+    return this;
+  }
+
+  includeExcludePaymentMethods(include: string[] = [], exclude: string[] = []) {
```
Comment (liorzam) [inline services/workflows-service/src/data-analytics/data-analytics.repository/analytics.query-builder.ts]: this is query build i suggest to include also field name as argument

File: services/workflows-service/package.json (hunk 1 above)
Comment (alonp99) [reply services/workflows-service/package.json:142]: yes

File: services/workflows-service/src/transaction/validations/create-transaction.validator.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import z from 'zod';
+import { PaymentMethodSchema } from '../schemas/zod-schemas';
+
+export const createTranscationValidator = z.object({
+  amount: z.number().optional(),
+  paymentMethod: PaymentMethodSchema.optional(),
+  currency: z.string().optional(),
+  description: z.string().optional(),
+  transactionId: z.string().optional(),
+  transactionType: z.enum(['Deposit', 'Withdrawal']),
```
Comment (liorzam) [inline services/workflows-service/src/transaction/validations/create-transaction.validator.ts:10]: Use prisma enum in order to keep transactionType up to date

**Pull request merged**